cql3/Cql.g

```diff
@@ -1621,7 +1621,7 @@ collectionLiteral returns [uexpression value]
 	@init{ std::vector<expression> l; }
     : '['
           ( t1=term { l.push_back(std::move(t1)); } ( ',' tn=term { l.push_back(std::move(tn)); } )* )?
-      ']' { $value = collection_constructor{collection_constructor::style_type::list, std::move(l)}; }
+      ']' { $value = collection_constructor{collection_constructor::style_type::list_or_vector, std::move(l)}; }
```

Comment:
> Motivation of using collection_constructor for representing vectors before preparation is that the vectorLiterals are indistinguishable from listLiterals before submiting any context of column type.

I think that the commit message could benefit from restructuring. You should start with motivation first, then explain what you are doing in the commit. It is much easier to read about the changes that you introduce when you have all the necessary context to understand _why_ you do them, rather than taking them at face value at first and later trying to fill the gaps in your understanding by reading the motivation. Apart from that, I think that this order generally leads to more natural and easier to understand text.

Considering that I recommend to split this into two commits - add `vector_constructor` + rename `list` to `list_or_vector` (I think this order makes most sense) - I think that the descriptions should follow this structure:

Add `vector_constructor`:
- Explain why it makes sense to have a separate `vector_costructor` from `collection_constructor`
- Note that the expression data structures can represent both typed and untyped expressions
- Explain that we cannot use the new variant for untyped expressions because the literal syntax for lists and vectors is exactly the same
- Explain that the variant is effectively unused as of this commit, but the next commit will adjust `list` to `list_or_vector` and will make use of the new `vector_constructor`

Rename `list` to `list_or_vector`:
- No need to provide lengthy motivation, it is sufficient to refer to the previous commit ("as explained in the previous commit, ..." etc.)
- Shortly sum up what is being done in this commit

Reply:
Proposed changes has been introduced during rebase.

Please check if the changes meet your requests.